test/csuite/wt7989_compact_checkpoint/main.c

```diff
@@ -0,0 +1,286 @@
+ * IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+ * OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+ * ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ * OTHER DEALINGS IN THE SOFTWARE.
+ */
+#include "test_util.h"
+
+/*
+ * This test executes to test cases: 
+ * - One with WT_TIMING_STRESS_CHECKPOINT_SLOW flag. It adds 10 secods sleep before each checkpoint.
+ * - Another test case synchronizes compact and checkpoint threads using a condition variable.
+ * The reason we have two test here is that they give different output when configured
+ * with "verbose=[compact,compact_progress]". There's a chance these two cases are different.
+ */
+
+#define NUM_RECORDS 1000000
+
+/* Constants and variables declaration. */
+/* 
+ * You may want to add "verbose=[compact,compact_progress]" to the connection config stringto get
+ * better view on what is happening.
+ */
+static const char conn_config[] =
+  "create,cache_size=2GB,statistics=(all)";
+static const char table_config[] =
+  "allocation_size=4KB,leaf_page_max=4KB,key_format=i,value_format=QQQS";
+static char data_str[1024] = "";
+
+/* Structures definition. */
+struct thread_data {
+    WT_CONNECTION *conn;
+    const char *uri;
+    WT_CONDVAR *cond;
+    WT_SESSION_IMPL *session_cond;
+};
+
+/* Forward declarations. */
+static void run_test(const char *home, const char *uri, bool stress_test);
+static void *thread_func_compact(void *arg);
+static void *thread_func_checkpoint(void *arg);
+static void populate(WT_SESSION *session, const char *uri);
+static void remove_records(WT_SESSION *session, const char *uri);
+static uint64_t get_file_size(WT_SESSION *session, const char *uri);
+static void set_timing_stress_checkpoint(WT_CONNECTION *conn);
+
+/* Methods implementation. */
+int
+main(int argc, char *argv[])
+{
+    TEST_OPTS *opts, _opts;
+    char home_cv[512];
+
+    opts = &_opts;
+    memset(opts, 0, sizeof(*opts));
+    testutil_check(testutil_parse_opts(argc, argv, opts));
+
+    /*
+     * First run test with WT_TIMING_STRESS_CHECKPOINT_SLOW.
+     */
+    run_test(opts->home, opts->uri, true);
+
+    /*
+     * Now run test where compact and checkpoint threads are synchronized using condition variable.
+     */
+    testutil_assert(sizeof(home_cv) > strlen(opts->home) + 3);
+    sprintf(home_cv, "%s.CV", opts->home);
+    run_test(home_cv, opts->uri, false);
+
+    /* Cleanup */
+    if (!opts->preserve)
+        testutil_clean_work_dir(home_cv);
+
+    testutil_cleanup(opts);
+
+    return (EXIT_SUCCESS);
+}
+
+static void
+run_test(const char *home, const char *uri, bool stress_test)
+{
+    struct thread_data td;
+    WT_CONNECTION *conn;
+    WT_SESSION *session;
+    pthread_t thread_checkpoint, thread_compact;
+    uint64_t file_sz_after, file_sz_before;
+
+    testutil_make_work_dir(home);
+    testutil_check(wiredtiger_open(home, NULL, conn_config, &conn));
+
+    if (stress_test) {
```

Comment:
You don't need the brackets here

Reply:
There's a multiline comment in the block. The brackets are there to improve readability.